File: Set/2215_Find_the_Difference_of_Two_Arrays/2215_Find_the_Difference_of_Two_Arrays.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        vector<vector<int>> answer;
        
        set<int> h1(nums1.begin(), nums1.end());
        set<int> h2(nums2.begin(), nums2.end());
        
        
        
        vector<int> v1;
        vector<int> v2;


        set_difference(h1.begin(), h1.end(), h2.begin(), h2.end(), inserter(v1, v1.begin()));
        set_difference(h2.begin(), h2.end(), h1.begin(), h1.end(), inserter(v2, v2.begin()));

        answer.push_back(v1);
        answer.push_back(v2);

        return answer;
    }
};

File: Set/2215_Find_the_Difference_of_Two_Arrays/2215_Find_the_Difference_of_Two_Arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        
        List<List<Integer>> answer = new ArrayList<List<Integer>>();

        Set<Integer> h1 = new HashSet<Integer>();
        for(int elem: nums1)
            h1.add(elem);
        
        Set<Integer> h2 = new HashSet<Integer>();
        for(int elem: nums2)
            h2.add(elem);

        Set<Integer> temp = new HashSet<>(h1);

        h1.removeAll(h2);
        h2.removeAll(temp);

        List<Integer> v1 = new ArrayList<>(h1);
        List<Integer> v2 = new ArrayList<>(h2);

        answer.add(v1);
        answer.add(v2);

        return answer;

    }
}